### ISSUE TITLE
Explicitly define ExternalName expectations

### ISSUE THREAD

**Issue by howardjohn** [kubernetes-sigs/gateway-api#1070]
Currently, Service is a supported backendRef. A service can be of type ExternalName, which has non-obvious semantics.

In the Ingress world, I previously did some research into how different controllers handle differently; I don't think I was able to find two implementations doing the same thing.

It would be great to explicitly define how this should be treated. Depending on the answer, we may need to more broadly address how we access external sites.

**Comment by youngnick** [comment]
I think we should recommend pretty strongly against using a Service of type ExternalName as a backendRef.

It's just too easy to bypass namespace controls by using ExternalNames of `service.namespace.svc.cluster.local`, or to gain access to the localhost of network gear using an ExternalName of `localhost`. And that's aside from the stuff you can do with manipulating Endpoints on top of that.

I'd rather see the creation of an explicit redirect-to-external-site resource that can explicitly rule out `localhost` and `cluster.local` domains.

**Comment by hbagdi** [comment]
> I think we should recommend pretty strongly against using a Service of type ExternalName as a backendRef.

I can see someone trying to use Gateway API as an Egress Gateway which specializes in doing routing to ExternalName services. No?

**Comment by youngnick** [comment]
Yes, I agree that is a useful use case, but allowing Service of Type ExternalName in all cases still leaves us with the ability to use this functionality to bypass namespace security. That's why I recommend a different object that can be clear about the risks involved.

The alternative is to allow Service of Type ExternalName in *some* use cases, but then there's no way for a user to know if their service will work or not. They'd have to create one and see. That doesn't seem to mesh well with the goal of increasing portability.
